pandas/tests/plotting/test_hist_method.py

```diff
@@ -169,6 +169,10 @@ def test_hist_df_legacy(self):
         with tm.assert_produces_warning(UserWarning):
             _check_plot_works(df.hist, bins=5)
 
+        # check bins argument with string
+        with tm.assert_produces_warning(UserWarning):
+            _check_plot_works(df.hist, bins="auto")
```

Comment:
Can you write a test that actually ensures the binning behavior of "auto", i.e. not just that this works without error but that it also actually provides back the correct bin values?

Reply:
So you want me to check that `np.histogram` works? Should I also check every possible valid string that `np.histogram` takes?

Reply:
Not asking to check `np.histogram` but rather that the plot renders the ticklabels of the rendered plot are as expected. I think can just use `get_xticklabels` to be sure

Reply:
Honestly not sure how to go about this other than manually creating a couple of histograms and just checking the locations of the bins. The xtick labels could be the same for a histogram that uses "auto" and one that doesn't. For example:

```
df = DataFrame(randn(1000))
df.hist(bins="auto")
```
![image](https://user-images.githubusercontent.com/5508833/76103995-9fe1c180-5fa0-11ea-884f-b96fd15b1d99.png)
```
df.hist()
```
![image](https://user-images.githubusercontent.com/5508833/76104009-a40ddf00-5fa0-11ea-8a95-4301753bb862.png)

Reply:
I think that's fine; even just checking for those labels would improve the test, knowing they don't stray from the default